### PR TITLE
feat(gov): proposal to set swap burner on mainnet and polygon

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This PR adds a proposal to set the swap burner contract correctly on mainnet and polygon

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by clemsos** [comment]
> NB: the mainnet address for swapburner is a placeholder for now

# Set UDT and SwapBurner in Unlock contract on Mainnet and Polygon


## Goal of the proposal

This proposal sets 1) the UDT address and 2) the Swap Burner contract address in the main Unlock factory contract on the following chains: Ethereum,Polygon.

## About this proposal

Now that UDT has been bridged, the new address has to be set in the main Unlock contract for the tokens to be distributed or swap/burned. For that, we call the `configUnlock` function with the bridge UDT address as parameter (keeping other parameters untouched).

A `SwapBurner` helper contract has been deployed on all three chains and will be added as setting to the main Unlock contract. This will enable the “swap and burn” feature of fees collected by the protocol on these chains, following the deployment of bridged versions of UDT there.

## How it works

The proposal uses a cross-chain proposal pattern that, once passed, will send the calls to multiple chains at once. This pattern has been introduced and tested in a [previous proposal](https://www.tally.xyz/gov/unlock/proposal/1926572528290918174819693611122933562560576845671089759587616947457423587439).

## The calls

This DAO proposal contains 2 calls:

### Ethereum (1) 1 call

  Changes sent to the Unlock contract at 0xe79B93f8E22676774F2A8dAd469175ebd00029FA

1. call `configUnlock` with the following parameters

  - _udt: 0x90DE74265a416e1393A450752175AED98fe11517
  - _weth: 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
  - _estimatedGasForPurchase: 200000
  - _symbol: UDT
  - _URI: https://locksmith.unlock-protocol.com/api/key/
  - _chainId: 1

2. call `setSwapBurner(0x52690873b22B0949A3A2c1AaD22653218460A002)` and set the [SwapBurner](https://etherscan.io/address/0x52690873b22B0949A3A2c1AaD22653218460A002) contract



### Polygon (137) 1 call

  Changes sent to the Unlock contract at 0xE8E5cd156f89F7bdB267EabD5C43Af3d5AF2A78f

1. call `configUnlock` with the following parameters

  - _udt: 0xf7E78d9C4c74df889A83C8C8d6D05BF70fF75876
  - _weth: 0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270
  - _estimatedGasForPurchase: 200000
  - _symbol: KEY
  - _URI: https://locksmith.unlock-protocol.com/api/key/137/
  - _chainId: 137

2. call `setSwapBurner(0x52690873b22B0949A3A2c1AaD22653218460A002)` and set the [SwapBurner](https://polygonscan.com/address/0x52690873b22B0949A3A2c1AaD22653218460A002) contract




The Unlock Protocol Team

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

File: governance/proposals/up/005-set-udt-swap-burner.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+## Goal of the proposal
+
+This proposal sets 1) the UDT address and 2) the Swap Burner contract address in the main Unlock factory contract on the following chains: ${targetChains
+    .map(({ name }) => name)
+    .toString()}. 
+  
+## About this proposal
+
+Now that UDT has been bridged, the new address has to be set in the main Unlock contract for the tokens to be distributed or swap/burned. For that, we call the \`configUnlock\` function with the bridge UDT address as parameter (keeping other parameters untouched).
```
Comment (ccarfi) [inline governance/proposals/up/005-set-udt-swap-burner.js:152]:
```suggestion
Now that UDT has been bridged, the new address has to be set in the main Unlock contract for the tokens to be distributed or swap/burned. For that, we call the \`configUnlock\` function with the bridge UDT address as a parameter (keeping other parameters untouched).
```

**Comment by ccarfi** [comment]
two VERY minor changes i made.

@clemsos quick question — this proposal mentions "UDT" and not "UPToken." i wanted to double-check that is correct.

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by unlock-deployer** [comment]
### packages/networks/src/networks/zksync.ts

The setup is successful :

- ✅ all mandatory keys are present
- ✅ Subgraph up and running
- ✅ Unlock ownership correctly set to the multisig







Some additonal warnings  :

- ⚠️ Skipping contract validation checks for packages/networks/src/networks/zksync.ts

**Comment by clemsos** [comment]
submitted here https://www.tally.xyz/gov/unlock-protocol/proposal/69517608692989012889331722369061426510780192090491006369616252524261713945600

**Review by julien51 — APPROVED** [review]

**Pull request merged**